### ISSUE TITLE
Motion detection, simpler example stand alone page

### ISSUE THREAD

**Issue by mrmccormack** [eclipsesource/tabris-js#1627]
### Problem description
- I want to make a single file tutorial on how to use Motion / Cordova plugin.  (and later, for all the plugins in /examples/cordova folder.)

- I find it hard to follow, when they are all combined.

**Request**
- Could anyone provide me some help on how to **combine** theses 3 pages  into one file for just Motion detection?

1. PluginPage.js
1. app.js
1. MotionPage.js

> Example Code:
https://github.com/eclipsesource/tabris-js/blob/master/examples/cordova/MotionPage.js

- I've tried, but I don't fully understand how the code works between the pages, and how to import the plugin...

- I did find: https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-device-motion/
 
- Thanks

### Environment
- Tabris.js version:  20.2.4
- Device: iPhone or Samsung
- OS: iOS 10.3.1 or Android 7.1

**Comment by cookieguru** [comment]
There's a lot of boilerplate between all the demo pages for all the plugins.  Rather than copy/paste that in to each demo page, the team has used a class to abstract that in to its own file.

As an example, there are some stylistic tweaks that happen.  On all pages, there is an element with ID `pluginIdLabel` and that gets its text color set on [line 45 of `PluginPage`](https://github.com/eclipsesource/tabris-js/blob/65820afb6cf63765ef19a6b248bf3ab9299f497c/examples/cordova/PluginPage.js#L45).  All the other plugin demo pages extend that, so when they are created, they also hit that line, by way of their parent's constructor.  The `SharingPage` also does the same thing, however it has an additional style that needs to be set when the page is created.  In [`SharingPage`](https://github.com/eclipsesource/tabris-js/blob/8806a5100bebbbca97090e91ddf1a7c6e3a89713/examples/cordova/SharingPage.js#L39) the method is overridden, but first calls the parent method to have those styles applied.  MDN has a great [document on ES6 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes).

In short, if you wanted it all in one file you could manually merge the contents of `MotionPage` in to `PluginPage` and then it will run on its own.

Alternatively, since the plugin has good docs, you can just copy/paste the snippets from their docs in to your app.  The only difference is that unlike with Cordova, you do *not* need to wait for the `deviceready` event to fire in Tabris.js.  The device is ready as soon as the first line of code executes.

**Comment by mrmccormack** [comment]
- **Thanks** for pointing me in the right direction
- it's far simpler than I was expecting...
- I didn't know how `deviceready` is handled in Tabrisjs.

---

### Well, it couldn't get much easier :smiley:
- here is my first attempt at getting motion detection working as a tutorial
- **bonus**  = it works in **Tabrisjs Playground**
 
```js
const { Button, TextView, ui } = require('tabris')

function onSuccess(acceleration) {
  txvAccelerationX.text = 'Acceleration X: ' + acceleration.x
  txvAccelerationY.text = 'Acceleration Y: ' + acceleration.y
  txvAccelerationZ.text = 'Acceleration Z: ' + acceleration.z
  txvTimeStamp.text = 'Timestamp: ' + getFormattedDate(acceleration.timestamp)
}

function onError() {
  console.log('onError!')
}

// updates every 1/4 second, 250 ms.

const options = { frequency: 250 }


let watchID = navigator.accelerometer.watchAcceleration(
  onSuccess,
  onError,
  options
)

let txvAccelerationX = new TextView({
  left: 10,
  top: 'prev() 10',
  right: 10,
  text: 'Acceleration X: ',
  alignment: 'left'
}).appendTo(ui.contentView)

let txvAccelerationY = new TextView({
  left: 10,
  top: 'prev() 10',
  right: 10,
  text: 'Acceleration Y: ',
  alignment: 'left'
}).appendTo(ui.contentView)

let txvAccelerationZ = new TextView({
  left: 10,
  top: 'prev() 10',
  right: 10,
  text: 'Acceleration Z: ',
  alignment: 'left'
}).appendTo(ui.contentView)

let txvTimeStamp = new TextView({
  left: 10,
  top: 'prev() 10',
  right: 10,
  text: 'Timestamp: ',
  alignment: 'left'
}).appendTo(ui.contentView)

function getFormattedDate () {
  let date = new Date()

  let month = date.getMonth() + 1
  let day = date.getDate()
  let hour = date.getHours()
  let min = date.getMinutes()
  let sec = date.getSeconds()

  month = (month < 10 ? '0' : '') + month
  day = (day < 10 ? '0' : '') + day
  hour = (hour < 10 ? '0' : '') + hour
  min = (min < 10 ? '0' : '') + min
  sec = (sec < 10 ? '0' : '') + sec

  let str =
    date.getFullYear() +
    '-' +
    month +
    '-' +
    day +
    ' » ' +
    hour +
    ':' +
    min +
    ':' +
    sec

  return str
}


```

**Comment by mpost** [comment]
That looks reasonable. Closing this issue for now.
